src/utils/transformForLoop.js

```diff
@@ -0,0 +1,54 @@
+import _ from 'lodash';
+import traverser from '../traverser';
+import copyComments from '../utils/copyComments';
+import {matchesAst} from '../utils/matchesAst';
+
+export function withComments(node, forEach) {
+  copyComments({from: node, to: forEach});
+  copyComments({from: node.body.body[0], to: forEach});
+  return forEach;
+}
+
+export function transformForLoopBodyElement({body, item, index, array}) {
+  return replaceArrayItemInBodyElement({
+    body: removeFirstBodyElement(body),
+    item,
+    index,
+    array
+  });
+}
+
+export function removeFirstBodyElement(body) {
+  return Object.assign({}, body, {
+    body: _.tail(body.body)
+  });
+}
+
+function replaceArrayItemInBodyElement({body, item, index, array}) {
+  traverser.replace(body, {
+    enter(node) {
+      const matches = matchesItem(index, array)(node);
+
+      if (matches) {
+        return item;
+      }
+    }
+  });
+
+  return body;
```

Comment:
Ah, I did miss that case. Can you provide any help on how you would use `isEqualAst()`?